substrate/runtime/council/src/motions.rs

```diff
@@ -0,0 +1,350 @@
+	type Proposal: Parameter + Dispatchable<Origin=<Self as Trait>::Origin> + MaybeSerializeDebug;
+
+	/// The outer event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+/// Origin for the system module. 
+#[derive(PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum Origin {
+	/// It has been condoned by a given number of council members.
+	Members(u32),
+}
+
+/// Outwardly visible event.
+pub type Event<T> = RawEvent<<T as system::Trait>::Hash, <T as system::Trait>::AccountId>;
+
+/// Event for this module.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, PartialEq, Eq, Clone)]
+pub enum RawEvent<Hash, AccountId> {
+	/// A motion (given hash) has been proposed (by given account) with a threshold (given u32).
+	Proposed(AccountId, Hash, u32),
+	/// A motion (given hash) has been voted on by given account, leaving
+	/// a tally (yes votes and no votes given as u32s respectively).
+	Voted(AccountId, Hash, bool, u32, u32),
+	/// A motion was approved by the required threshold.
+	Approved(Hash),
+	/// A motion was not approved by the required threshold.
+	Disapproved(Hash),
+	/// A motion was executed; `bool` is true if returned without error.
+	Executed(Hash, bool),
+}
+
+impl<H, A> From<RawEvent<H, A>> for () {
+	fn from(_: RawEvent<H, A>) -> () { () }
+}
+
+decl_module! {
+	#[cfg_attr(feature = "std", serde(bound(deserialize = "<T as Trait>::Proposal: ::serde::de::DeserializeOwned")))]
+	pub struct Module<T: Trait> for enum Call where origin: <T as system::Trait>::Origin {
+		fn propose(origin, threshold: u32, proposal: Box<<T as Trait>::Proposal>) -> Result;
+		fn vote(origin, proposal: T::Hash, approve: bool) -> Result;
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as CouncilMotions {
+		/// The (hashes of) the active proposals.
+		pub Proposals get(proposals): default Vec<T::Hash>;
+		/// Actual proposal for a given hash, if it's current.
+		pub ProposalOf get(proposal_of): map [ T::Hash => <T as Trait>::Proposal ];
+		/// Votes for a given proposal: (required_yes_votes, yes_voters, no_voters).
+		pub Voting get(voting): required map [ T::Hash => (u32, Vec<T::AccountId>, Vec<T::AccountId>) ];
+	}
+}
+
+impl<T: Trait> Module<T> {
+
+	/// Deposit one of this module's events.
+	fn deposit_event(event: Event<T>) {
+		<system::Module<T>>::deposit_event(<T as Trait>::Event::from(event).into());
+	}
+
+	pub fn is_councillor(who: &T::AccountId) -> bool {
+		<Council<T>>::active_council().iter()
+			.any(|&(ref a, _)| a == who)
+	}
+
+	// Dispatch
+	fn propose(origin: <T as system::Trait>::Origin, threshold: u32, proposal: Box<<T as Trait>::Proposal>) -> Result {
+		let who = ensure_signed(origin)?;
+
+		ensure!(Self::is_councillor(&who), "proposer not on council");
+
+		let proposal_hash = T::Hashing::hash_of(&proposal);
+
+		ensure!(!<ProposalOf<T>>::exists(proposal_hash), "duplicate proposals not allowed");
```

Comment:
note that there is a (slight) chance of reusing votes.

Let's say that we have proposal P, 4 active councillors (c0..c3) and the following extrinsics were published:

1. `propose(c0, 2, P)`
2. `vote(c1, H(P))`
3. `vote(c2, H(P))`
4. `vote(c3, H(P))`

Now it seems that the 3rd extrinsic [will panic](
https://github.com/paritytech/substrate/pull/694/files#r216255574), but let's assume that it's fixed by no-oping. 

Some rogue councillor `c0` can issue `propose(c0, 2, P)` extrinsic that has a slight chance to be applied before votes from `c2` and `c3` and thus use these votes for approving the second proposal. This would be more effective, if `c0` also act as front-running validator.

Reply:
i guess i can introduce a proposal index to ensure they're unique.